test/models/compute_resource_test.rb

```diff
@@ -383,4 +383,17 @@ def setup
       refute cr.capable?(:build)
     end
   end
+
+  test "returns nested_attribute_for ActionController::Parameters" do
+    cr = compute_resources(:mycompute)
+    ActionController::Parameters.permit_all_parameters = true
+    hash = {:disk => "test"}
+
+    volume_attributes = ActionController::Parameters.new("1520857914238" => ActionController::Parameters.new(hash))
+    volumes = cr.send(:nested_attributes_for, :volumes, volume_attributes)
+
+    ActionController::Parameters.permit_all_parameters = false
```

Comment:
I think this should be done in an `ensure` block to prevent side effects if something fails earlier, or instead of using `permit_all_parameters` specifically permit the parameters for this AC params to simulate the way it would work normally.